engine/gui/src/gui.h

```diff
@@ -422,15 +422,16 @@ namespace dmGui
      * @param buffer
      * @param context
      */
-    typedef void* (*NewTexture)(HScene scene, uint32_t width, uint32_t height, dmImage::Type type, const void* buffer, void* context);
+    typedef HTextureSource (*NewTexture)(HScene scene, uint32_t width, uint32_t height, dmImage::Type type, const void* buffer, void* context);
 
     /**
      * Delete texture callback
      * @param scene
      * @param texture
+     * @param type
      * @param context
      */
-    typedef void (*DeleteTexture)(HScene scene, void* texture, void* context);
+    typedef void (*DeleteTexture)(HScene scene, HTextureSource texture, NodeTextureType type, void* context);
```

Comment:
I felt it was a bit weird that the function is _only_ used for dynamic textures, but we don't have anything in the function that tells us that.. shouldn't it be called DeleteDynamicTexture?

Reply:
Yeah, that is probably a good idea as well.